sdk/storage/inc/datalake/path_client.hpp

```diff
@@ -0,0 +1,285 @@
+
+    /**
+     * @brief Sets POSIX access control rights on files and directories under given path
+     *        recursively.
+     * @param mode Mode PathSetAccessControlRecursiveMode::Set sets POSIX access control rights on
+     *             files and directories, PathSetAccessControlRecursiveMode::Modify modifies one or
+     *             more POSIX access control rights  that pre-exist on files and directories,
+     *             PathSetAccessControlRecursiveMode::Remove removes one or more POSIX access
+     *             control rights that were present earlier on files and directories
+     * @param options Optional parameters to set an access control recursively to the resource the
+     *                path points to.
+     * @return PathSetAccessControlRecursiveResponse
+     */
+    PathSetAccessControlRecursiveResponse SetAccessControlRecursive(
+        PathSetAccessControlRecursiveMode mode,
+        const SetAccessControlRecursiveOptions& options = SetAccessControlRecursiveOptions()) const;
+
+    /**
+     * @brief Sets the properties of a resource the path points to.
+     * @param options Optional parameters to set the properties to the resource the path points to.
+     * @return PathUpdateResponse
+     */
+    PathUpdateResponse SetProperties(
+        const SetPathPropertiesOptions& options = SetPathPropertiesOptions()) const;
+
+    /**
+     * @brief Uploads data to be appended to a file, flushes (writes) previously uploaded data to a
+     *        file, sets properties for a file or directory, or sets access control for a file or
+     *        directory. Data can only be appended to a file.
+     * @param action The action must be PathUpdateAction::Append to upload data to be appended to a
+     *               file, PathUpdateAction::Flush to flush previously uploaded data to a file,
+     *               PathUpdateAction::SetProperties to set the properties of a file or directory,
+     *               or PathUpdateAction::SetAccessControl to set the owner, group, permissions, or
+     *               access control list for a file or directory. Note that Hierarchical Namespace
+     *               must be enabled for the account in order to use access control. Also note that
+     *               the Access Control List (ACL) includes permissions for the owner, owning group,
+     *               and others, so the x-ms-permissions and x-ms-acl request headers are mutually
+     *               exclusive.
+     * @param options Optional parameters to update the resource the path points to.
+     * @return UpdatePathResponse
+     * @remark This API is not advised to be used unless absolutely needed. Please use AppendData,
+     *         FlushData, SetAccessControl, SetAccessControlRecursive, SetProperties instead for
+     *         specific functionality.
+     */
+    UpdatePathResponse Update(
+        PathUpdateAction action,
+        const PathUpdateOptions& options = PathUpdateOptions()) const;
```

Comment:
No, `Update` is a bad name. We should split it into several functions.

Reply:
It's already split into several functions. Update is left here for REST parity.

Reply:
Then I don't think this function is necessary any more. Does JS SDK or .Net SDK have such a function in convenience layer?

Reply:
Just checked, there isn't such an API in .NET sdk